ui/app/actions.js

```diff
@@ -746,20 +745,26 @@ function updateGasData ({
 }) {
   return (dispatch) => {
     dispatch(actions.gasLoadingStarted())
-    const estimatedGasPrice = estimateGasPriceFromRecentBlocks(recentBlocks)
-    return Promise.all([
-      Promise.resolve(estimatedGasPrice),
-      estimateGas({
+    let gasPrice
```

Comment:
Why is `let gasPrice` needed here?

Reply:
We need the value resolved by the first promise, it then needs to be passed to estimateGas which returns another promise, and then is needed again after estimateGas resolves.

Reply:
As an alternative, the old form doesn't require external state (i.e. using `Promise.resolve(gasPrice)` to pass the gas price through the promise chain):

```js
return new Promise((resolve, reject) => {
  background.getGasPrice((err, data) => {
    if (err !== null) return reject(err)
    return resolve(data)
  })
})
.then(estimateGasPrice => {
  return Promise.all([
    Promise.resolve(estimateGasPrice),
    estimateGas({
      estimateGasMethod: background.estimateGas,
      blockGasLimit,
      selectedAddress,
      selectedToken,
      to,
      value,
      gasPrice,
    })
  ])
})
.then(([gasPrice, gas]) => {
  dispatch(actions.setGasPrice(gasPrice))
  dispatch(actions.setGasLimit(gas))
  return calcGasTotal(gas, gasPrice)
})
```